### DISCUSSION TITLE
Retrieve Messages

### DISCUSSION THREAD

**Discussion by smartos99** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#63] [Q&A] [answered]
Hi,
If I have the wamid, Is there a way to retrieve message info including its content? 

Thank you in advance.

**Comment by gabrieldwight** [comment, marked as answer]
Hi, there is no available api to retrieve the message content with wamid. The only way to get the content is when you are receiving the messages which the contents will be sent to the webhook.

**Comment by theCJMan** [comment]
My question is possibly related to this one.
If my app goes down, and after some time comes back up, is there a way to get ALL messages on the server?

If not, what happens when the app is down for sometime and only come backup after several hours...
Are those received messages (on the Server) lost?

**Reply by gabrieldwight** [reply]
The cloud API will retry to send the received message ID to the server if your server does not return a 200 status code but has its retry policies and will depend on how many received messages it can send your server.

**Comment by theCJMan** [comment]
I understand for be messages.
Is there any way to get historical messages from the server.
Let's say my server died and needs to rebuild the server.
Can i then get historical messages, or will these historical messages be lost


⁣Get BlueMail for Android ​

**Reply by gabrieldwight** [reply]
The historical messages cannot be retrieved since the cloud API does not have any endpoint for retrieving historical messages. The other way is to store the messages received in your database but you can use serverless functions to handle the webhook data.
